src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -50,27 +53,45 @@ public class FileService implements DisposableBean {
 
     private final ScheduledExecutorService executor = Executors.newScheduledThreadPool(Runtime.getRuntime().availableProcessors());
 
+    // NOTE: this list has to be the same as in file-uploader.service.ts
+    private final List<String> allowedFileExtensions = new ArrayList<>(Arrays.asList("png", "jpg", "jpeg", "svg", "pdf", "zip"));
+
+    public static final String MARKDOWN_FILE_SUBPATH = "/api/files/markdown/";
+
+    public static final String DEFAULT_FILE_SUBPATH = "/api/files/temp/";
+
     /**
      * Filenames for which the template filename differs from the filename it should have in the repository.
      */
     // @formatter:off
-    private static final Map<String, String> FILENAME_REPLACEMENTS = Map.ofEntries(
-        Map.entry("git.ignore.file", ".gitignore"),
-        Map.entry("git.attributes.file", ".gitattributes"),
-        Map.entry("Makefile.file", "Makefile"),
-        Map.entry("dune.file", "dune"),
-        Map.entry("Fast.file", "Fastfile"),
-        Map.entry("App.file", "Appfile"),
-        Map.entry("Scan.file", "Scanfile"),
-        Map.entry("gradlew.file", "gradlew")
-    );
+    private static final Map<String, String> FILENAME_REPLACEMENTS = Map.ofEntries(Map.entry("git.ignore.file", ".gitignore"), Map.entry("git.attributes.file", ".gitattributes"),
```

Comment:
Your IDE probably ignored the `@formatter:off` comment here. They were formatted manually before to have a single entry per line. I think it would be nice to keep that as it makes the list a lot more readable.